src/Features/Core/Portable/Completion/Providers/SymbolCompletionItem.cs

```diff
@@ -30,26 +30,29 @@ internal static class SymbolCompletionItem
             IReadOnlyList<ISymbol> symbols,
             CompletionItemRules rules,
             int contextPosition,
-            Action<IReadOnlyList<ISymbol>, ImmutableDictionary<string, string>.Builder> symbolEncoder,
+            Action<IReadOnlyList<ISymbol>, ArrayBuilder<KeyValuePair<string, string>>> symbolEncoder,
             string? sortText = null,
             string? insertionText = null,
             string? filterText = null,
             SupportedPlatformData? supportedPlatforms = null,
-            ImmutableDictionary<string, string>? properties = null,
+            ImmutableArray<KeyValuePair<string, string>> properties = default,
             ImmutableArray<string> tags = default,
             string? displayTextPrefix = null,
             string? inlineDescription = null,
             Glyph? glyph = null,
             bool isComplexTextEdit = false)
         {
-            var builder = properties != null ? properties.ToBuilder() : ImmutableDictionary<string, string>.Empty.ToBuilder();
+            using var _ = ArrayBuilder<KeyValuePair<string, string>>.GetInstance(out var builder);
```

Comment:
nit: temporary array.

Reply:
Couldn't find a way to easily switch over this one since the symbolEncoder parameter needs to add to the collection.